autoreject/tests/test_ransac.py

```diff
@@ -50,12 +53,146 @@ def test_ransac():
+                      1/epochs.info['sfreq'])
+    sinewave = 10 * np.sin(2*np.pi*50*times)
+        
+    epochs._data[:,noisy_idx,:] *= sinewave
+   
+    
+    # atomic tests for _get_mappings to check if they return the same
+    # result when run "parallel" for subsets or all-in-once, sanity check
+    ransac = Ransac(picks=mag_idx, random_state=np.random.RandomState(42),
+                        n_jobs=1, n_resample=2, min_channels=0.1)
+    ch_subset = ransac._get_random_subsets(epochs.info)
+    ransac.ch_type='meg'
+    mappings = ransac._get_mappings(epochs, ch_subset)
+    mappings_sub1 = ransac._get_mappings(epochs, [ch_subset[0]])
+    mappings_sub2 = ransac._get_mappings(epochs, [ch_subset[1]])
+    
+    np.testing.assert_array_equal(mappings, np.concatenate([mappings_sub1,
+                                                          mappings_sub2]))
+    # atomic tests for _iterate_epochs to check if they return the same
+    # result when run "parallel" for subsets or all-in-once, sanity check
+    ransac.mappings_ = mappings
+    epoch_idxs_splits = np.array_split(np.arange(len(epochs)), 2)
+    corr = ransac._iterate_epochs(epochs, np.arange(len(epochs)))
+    corr_sub1 = ransac._iterate_epochs(epochs, epoch_idxs_splits[0])
+    corr_sub2 = ransac._iterate_epochs(epochs, epoch_idxs_splits[1])
+    np.testing.assert_array_equal(corr, np.concatenate([corr_sub1,
+                                                        corr_sub2]))
+
+    # now test across different jobs
+    mappings = {}
+    corrs = {}
+    bad_chs = {}
+    
+    for n_jobs in [1, 2, -1]:
+        
+        ransac = Ransac(picks=mag_idx, random_state=np.random.RandomState(42),
+                        n_jobs=n_jobs, n_resample=50)
+        ransac.fit(epochs)
+        
+        # the corr_ variable should be of shape (n_epochs, n_channels)
+        assert ransac.corr_.shape==(len(epochs), len(mag_idx))
+        corrs[n_jobs] = ransac.corr_
+        bad_chs[n_jobs] = ransac.bad_chs_
+        mappings[n_jobs] = ransac.mappings_
+        
+        # should find all noisy bad channels
+        bads = [epochs.ch_names.index(ch) for ch in ransac.bad_chs_]
+        for noisy in noisy_idx:
+            assert noisy in bads, \
+                f'bad channel {noisy} not detected, bads={bads}'
+        
+    # test if results are consistent across different jobs
+    for n_jobs in corrs:
+        # simply compare to the last computed value
+        np.testing.assert_allclose(mappings[1], mappings[n_jobs], err_msg=
+                f'n_jobs={n_jobs} did not produce same results as non-parallel')
+        np.testing.assert_allclose(corrs[1], corrs[n_jobs], err_msg=
+                f'n_jobs={n_jobs} did not produce same results as non-parallel')
+        assert bad_chs[1]==bad_chs[n_jobs], \
+                f'n_jobs={n_jobs} did not produce same results as non-parallel'
+
+def test_ransac_backwards_equality():
```

Comment:
I think this test is not needed as long as you've checked that the result doesn't change for n_jobs = 1

Reply:
Which one do you mean exactly? the test for same `bad_chs` being found when using different `n_jobs`? Or the backwards equality?

Reply:
I removed the backwards check again, but now you know it was there, and it was indeed consistent :)